contrib/epee/include/hex.h

```diff
@@ -68,4 +68,12 @@ namespace epee
     //! Write `src` bytes as hex to `out`. `out` must be twice the length
     static void buffer_unchecked(char* out, const span<const std::uint8_t> src) noexcept;
   };
+
+  struct from_hex
+  {
+      //! \return An std::vector of unsigned integers from the `src`
+      // would be nice to use a gsl::string_span here, but that one
+      // needs c++14 or needs to be backported like gsl::span
+      static std::vector<uint8_t> vector(const std::string &src);
```

Comment:
There is a boost::string_span which we use in a few places.

Reply:
I couldn't find any mention of string_span in the repo, I did find a ton of mentions for string_ref though, so I could use that indeed. That raises the issue that boost::string_ref is more or less deprecated (see https://lists.boost.org/Archives/boost/2017/07/236903.php ). The string_ref author recomments switching to (boost|std)::string_view.

It's probably best to use boost::string_ref for now and maybe change all references to boost::string_ref to boost::string_view in another PR - it's a trivial change.

Reply:
Sorry, I did mean boost::string_ref indeed.